### PR TITLE
[OID4VCI]: Enforce batch_size ≥ 2 validation for batch_credential_issuance

### LINKED ISSUE #77: Keycloak: Update batch_credential_issuance to enforce batch_size ≥ 2.

### PR DESCRIPTION
**Pull request by forkimenjeckayang**
## Summary
This PR implements validation for the `batch_credential_issuance.batch_size` parameter to ensure compliance with the OID4VCI specification, which requires that `batch_size` MUST be 2 or greater.

### Implementation
- Added validation in `getBatchCredentialIssuance()` method to check that `batch_size` is at least 2

### Testing
- Added comprehensive test method `testBatchCredentialIssuanceValidation()` covering:
    - Valid cases: batch sizes 2, 5 (should be accepted)
    - Invalid cases: batch sizes -1, 0, 1 (should be rejected)
    - Edge case: batch size exactly 2 (minimum valid value)
  - Created reusable helper method `testBatchSizeValidation()` to reduce code duplication
  - Ensured tests don't interfere with each other by properly restoring original realm attributes

## Addresses
- Issue https://github.com/adorsys/keycloak-oid4vc/issues/77

### CODE REVIEW

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oid4vc/issuance/signing/OID4VCIssuerWellKnownProviderTest.java [hunk 1]
```diff
@@ -413,6 +413,60 @@ public static void testCredentialConfig(SuiteContext suiteContext, KeycloakTesti
...
+                        // Restore original value
+                        if (originalBatchSize != null) {
+                            realm.setAttribute("batch_credential_issuance.batch_size", originalBatchSize);
+                        } else {
+                            realm.removeAttribute("batch_credential_issuance.batch_size");
+                        }
+                    }
+                });
+    }
+
```
Comment (Ogenbertrand) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oid4vc/issuance/signing/OID4VCIssuerWellKnownProviderTest.java:467]: I dont really think this test is necessary Dear @forkimenjeckayang as the exiting tests will behave the same and obvoiusly failed in case the batch size are less than the expected. Please double check the existing tests to confirm that what i am saying is right. Let's avoid repeatitive tests in the code base.

Comment (forkimenjeckayang) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oid4vc/issuance/signing/OID4VCIssuerWellKnownProviderTest.java:467]: You're right @Ogenbertrand 
The other tests will fail when `batch_size` doesn't have the required value. Just explicity added this  to test all scenarious. So you suggest i remove this?

File: services/src/main/java/org/keycloak/protocol/oid4vc/issuance/OID4VCIssuerWellKnownProvider.java [hunk 2]
```diff
@@ -95,8 +95,13 @@ private CredentialIssuer.BatchCredentialIssuance getBatchCredentialIssuance(Keyc
         String batchSize = realm.getAttribute("batch_credential_issuance.batch_size");
         if (batchSize != null) {
```
Comment (IngridPuppet) [inline services/src/main/java/org/keycloak/protocol/oid4vc/issuance/OID4VCIssuerWellKnownProvider.java]: Can this constant be added to Oid4vcConstants?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oid4vc/issuance/signing/OID4VCIssuerWellKnownProviderTest.java [hunk 3]
```diff
@@ -413,6 +413,60 @@ public static void testCredentialConfig(SuiteContext suiteContext, KeycloakTesti
...
+                            Assert.assertNull("batch_credential_issuance should be null for invalid batch size " + batchSize,
+                                    issuer.getBatchCredentialIssuance());
+                        }
+                    } finally {
+                        // Restore original value
+                        if (originalBatchSize != null) {
+                            realm.setAttribute("batch_credential_issuance.batch_size", originalBatchSize);
+                        } else {
+                            realm.removeAttribute("batch_credential_issuance.batch_size");
+                        }
```
Comment (IngridPuppet) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oid4vc/issuance/signing/OID4VCIssuerWellKnownProviderTest.java]: This is generally a weak pattern because if this isn't run for some reason, the state is not restored and another unrelated may fail unexpectedly. Those situations could be very hard to debug and should ideally be avoided.

To make the method more testable, you could rewrite it as:

```java
public static CredentialIssuer.BatchCredentialIssuance getBatchCredentialIssuance(RealmModel realm) {
```

By accepting a realm model instead of the Keycloak session, it should be straightforward to create new realms on every call and not alter the shared state.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oid4vc/issuance/signing/OID4VCIssuerWellKnownProviderTest.java (hunk 1 above)
Comment (IngridPuppet) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oid4vc/issuance/signing/OID4VCIssuerWellKnownProviderTest.java:467]: > The other tests will fail when batch_size doesn't have the required value.

I'm curious as to why they would fail. My understanding of the code is that the config is silently dismissed if invalid.

**Review by IngridPuppet — COMMENTED** [review]
Some minor comments, please could you check?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oid4vc/issuance/signing/OID4VCIssuerWellKnownProviderTest.java (hunk 1 above)
Comment (forkimenjeckayang) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oid4vc/issuance/signing/OID4VCIssuerWellKnownProviderTest.java:467]: As last consideration.
@IngridPuppet , you're right, the existing tests won't catch validation failures because they use valid values.
And i'l go ahead to keep the test, it's actually testing a different aspect of the functionality that existing tests don't cover.

**Review by IngridPuppet — APPROVED** [review]
LGTM
